thesis-api-v2/src/main/java/pl/edu/agh/quizzesthesis/api/dto/QuestionResponse.java

```diff
@@ -0,0 +1,6 @@
+package pl.edu.agh.quizzesthesis.api.dto;
+
+import java.util.List;
+
+public record QuestionResponse(int type, String question, TermResponse correct, List<String> answers,
+                               AdditionalInfoResponse additionalInfo) {}
```

Comment:
I was thinking "We do not know what exactly `additionalInfo` is for given `type` of question, it may be `authorName`, `authorProfileURL`, `null`, `something else` and we should be able to cover every case" so `AdditionalInfo` should be something generic, for example `Object class`. I tested it with frontend and after many failures I managed to properly display `authorName` with that change.

Reply:
Also, `type` shouldn't be `int` but we discussed it already and decided to go with it and change it in the future.